### ISSUE TITLE
TimeoutStrategy.Agressive causes TimeoutPolicy to time out immediately 

### ISSUE THREAD

**Issue by ekillops** [connor4312/cockatiel#16]
**TimeoutPolicy.ts (39)**

```
  public async execute<T>(fn: (context: ICancellationContext) => PromiseLike<T> | T): Promise<T> {
    const cts = new CancellationTokenSource();
    const timer = setTimeout(() => cts.cancel(), this.duration);

    try {
      if (this.strategy === TimeoutStrategy.Cooperative) {
        return await fn({ cancellation: cts.token });
      }

      return Promise.race<T>([
        fn({ cancellation: cts.token }),
        cts.token.cancellation(cts.token).then(() => {
          throw new TaskCancelledError(`Operation timed out after ${this.duration}ms`);
        }),
      ]);
    } finally {
      cts.cancel();
      clearTimeout(timer);
    }
  }
```

When using `TimeoutStrategy.Aggressive`, the `Promise.race` call is not `await`-ed causing the `finally` block to execute immediately, resulting in  `cts.cancel()` being called before the `timer` runs out. 

In my testing, adding an `await` before the `Promise.race` call results in the expected behavior of the cancellation being triggered from within the `setTimeout` call.

**Comment by connor4312** [comment]
Good catch!

**Comment by ekillops** [comment]
PR submitted.

**Comment by connor4312** [comment]
Thanks for the contribution, published in 0.1.4 ✨

**Comment by ekillops** [comment]
Sweet! Thanks for the fast response.
